sdk/openai/azure-ai-openai/src/test/java/com/azure/ai/openai/generated/OpenAIAsyncClientTest.java

```diff
@@ -0,0 +1,151 @@
+                })
+                .verifyComplete();
+        });
+    }
+
+    @ParameterizedTest(name = DISPLAY_NAME_WITH_ARGUMENTS)
+    @MethodSource("com.azure.ai.openai.generated.TestUtils#getTestParameters")
+    public void getChatCompletions(HttpClient httpClient, OpenAIServiceVersion serviceVersion) {
+        client = getOpenAIAsyncClient(httpClient, serviceVersion);
+        getChatCompletionsRunner((deploymentId, chatMessages) -> {
+            StepVerifier.create(client.getChatCompletions(deploymentId, new ChatCompletionsOptions(chatMessages)))
+                .assertNext(resultChatCompletions -> {
+                    assertNotNull(resultChatCompletions.getUsage());
+                    // TODO: assert chat completion and make it generic
+                    List<ChatChoice> choices = resultChatCompletions.getChoices();
+                    assertNotNull(choices);
+                    assertTrue(choices.size() > 0);
+                    ChatChoice chatChoice = resultChatCompletions.getChoices().get(0);
+                    assertEquals(0, chatChoice.getIndex());
+                    assertEquals(ChatRole.ASSISTANT, chatChoice.getMessage().getRole());
+                    assertNotNull(chatChoice.getMessage().getContent());
+                })
+                .verifyComplete();
+        });
+    }
+
+    @ParameterizedTest(name = DISPLAY_NAME_WITH_ARGUMENTS)
+    @MethodSource("com.azure.ai.openai.generated.TestUtils#getTestParameters")
+    public void getChatCompletionsWithResponse(HttpClient httpClient, OpenAIServiceVersion serviceVersion) {
+        client = getOpenAIAsyncClient(httpClient, serviceVersion);
+        getChatCompletionsRunner((deploymentId, chatMessages) -> {
+            StepVerifier.create(client.getChatCompletionsWithResponse(deploymentId,
+                    BinaryData.fromObject(new ChatCompletionsOptions(chatMessages)),
+                    new RequestOptions()))
+                .assertNext(response -> {
+                    assertEquals(200, response.getStatusCode());
+                    ChatCompletions resultChatCompletions = response.getValue().toObject(ChatCompletions.class);
+                    assertNotNull(resultChatCompletions.getUsage());
+                    // TODO: assert chat completion and make it generic
+                    List<ChatChoice> choices = resultChatCompletions.getChoices();
+                    assertNotNull(choices);
+                    assertTrue(choices.size() > 0);
+                    ChatChoice chatChoice = resultChatCompletions.getChoices().get(0);
+                    assertEquals(0, chatChoice.getIndex());
+                    assertEquals(ChatRole.ASSISTANT, chatChoice.getMessage().getRole());
+                    assertNotNull(chatChoice.getMessage().getContent());
+                })
+                .verifyComplete();
+        });
+    }
+
+    @Disabled("Status code 401, Access denied due to invalid subscription key or wrong API endpoint")
```

Comment:
Why is this disabled?

Reply:
enabled. In the shared openai azure resource, The previous embedding model is no longer accessible. I deploy another embedding model and am able to test it now.